### PR TITLE
QDM Model generator script

### PR DESCRIPTION
**Pull request by holmesie**
Initial script to generate Ruby (mongoid) and JavaScript (mongoose) models that correspond to the QDM spec as defined in a modelinfo XML file.

### CODE REVIEW

File: app/models/types/code.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  def mongoize
+    [ code, code_system, descriptor, code_system_oid, version ]
+  end
+
+  class << self
+
+    # Get the object as it was stored in the database, and instantiate
+    # this custom class from it.
+    def demongoize(object)
+      Code.new(object[0], object[1], object[2], object[3], object[4])
```
Comment (okeefm) [inline app/models/types/code.rb]: Do you want to add something to the comment for this that says "The array indices in `demongoize` are the same 5 elements used in `mongoize`" or the like?

File: app/models/types/interval.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  # Converts an object of this instance into a database friendly value.
+  def mongoize
+    [ lt, gt ]
+  end
+
+  class << self
+
+    # Get the object as it was stored in the database, and instantiate
+    # this custom class from it.
+    def demongoize(object)
```
Comment (okeefm) [inline app/models/types/interval.rb]: Same with `interval`, we should clarify the inputs to `demongoize` (in a comment is fine)

File: app/models/types/quantity.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  # Converts an object of this instance into a database friendly value.
+  def mongoize
+    [ value, unit ]
+  end
+
+  class << self
+
+    # Get the object as it was stored in the database, and instantiate
+    # this custom class from it.
+    def demongoize(object)
```
Comment (okeefm) [inline app/models/types/quantity.rb]: Same as `interval` and `code`.

File: app/models/types/code.rb (hunk 1 above)
Comment (holmesie) [reply app/models/types/code.rb]: Yes, I think that clarification make sense. Added.

File: app/models/types/interval.rb (hunk 2 above)
Comment (holmesie) [reply app/models/types/interval.rb]: Agreed.

File: app/models/types/quantity.rb (hunk 3 above)
Comment (holmesie) [reply app/models/types/quantity.rb]: Agreed.

**Comment by holmesie** [comment]
@okeefm In reply to your question about `code/interval/quantity`. Each of these types were a little more complicated than the standard types included by mongoid/mongoose, so I implemented them myself for use as datatype attribute types (so they aren't actually QDM datatypes, but are sort of primitives used as datatype attribute types).

**Pull request merged**